### PR TITLE
Fixed MA software crash on startup 

### PR DESCRIPTION
**Pull request by NikitaPoly**
Specific screen resolutions caused incorrect parsing of numbers.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -59,6 +59,7 @@ __pycache__/
 # C/C++ libraries
 *.zip
 src/zlib
+src/zlib-1.2.12
```
Comment (tannermpeterson) [inline .gitignore:62]: src/zlib* might work instead of adding this line

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[build-system]
```
Comment (tannermpeterson) [inline pyproject.toml:1]: I think you can delete this whole file and remove it from the .gitignore

File: src/main/BrowserWinHandler.js [hunk 3]
```diff
@@ -36,67 +36,84 @@ export default class BrowserWinHandler {
...
   _create() {
-    // In order to get display size to match in windows: https://stackoverflow.com/questions/59385237/electron-window-dimensions-vs-screen-resolution
-    const scale_factor = screen.getPrimaryDisplay().scaleFactor;
-    console.log("Screen size scale factor: " + scale_factor); // allow-log
-    const { width, height } = screen.getPrimaryDisplay().workAreaSize;
-    console.log("Sceen work area width " + width + " height " + height); // allow-log
-    this.options.height = this.options.height / scale_factor;
-    this.options.width = this.options.width / scale_factor;
-    this.options.webPreferences.zoomFactor = this.options.webPreferences.zoomFactor / scale_factor;
+      // In order to get display size to match in windows: https://stackoverflow.com/questions/59385237/electron-window-dimensions-vs-screen-resolution
```
Comment (tannermpeterson) [inline src/main/BrowserWinHandler.js:39]: I think you'll want to fix the indentation here or one of the pre-commit checks will fail. You can run pre-commit outside of an actual git commit with `pipenv run pre-commit run -a`

File: src/main/BrowserWinHandler.js [hunk 4]
```diff
@@ -36,67 +36,84 @@ export default class BrowserWinHandler {
...
-    this.browserWindow.on("close", async (e) => {
-      if (!close) {
-        e.preventDefault();
-        this.browserWindow.webContents.send("confirmation_request");
+      this.browserWindow.setContentSize(this.options.width, this.options.height); // Eli (6/20/20): for some odd reason, in Windows the EXE is booting up at slightly smaller dimensions than defined, so seeing if this extra command helps at all
+      // Eli (6/21/20): the position seems to be defaulting to 1,87 instead of the 0,0 supplied in the args, unless this extra method is called
+      this.browserWindow.setPosition(this.options.x, this.options.y);
+      const win_position = this.browserWindow.getPosition();
+      console.log("actual window position: " + win_position); // allow-log
+      try {
```
Comment (tannermpeterson) [inline src/main/BrowserWinHandler.js:90]: We should remove all these try catch blocks we added

File: src/main/BrowserWinHandler.js [hunk 5]
```diff
@@ -41,8 +41,8 @@ export default class BrowserWinHandler {
     console.log("Screen size scale factor: " + scale_factor); // allow-log
     const { width, height } = screen.getPrimaryDisplay().workAreaSize;
     console.log("Sceen work area width " + width + " height " + height); // allow-log
-    this.options.height = this.options.height / scale_factor;
-    this.options.width = this.options.width / scale_factor;
+    this.options.height = parseInt(this.options.height / scale_factor);
+    this.options.width = parseInt(this.options.width / scale_factor);
     this.options.webPreferences.zoomFactor = this.options.webPreferences.zoomFactor / scale_factor;
```
Comment (tannermpeterson) [inline src/main/BrowserWinHandler.js:46]: Last thing, we should also add parseInt here

Comment (NikitaPoly) [reply src/main/BrowserWinHandler.js:46]: Thought I added that. Will check my branches next time.

Comment (tannermpeterson) [reply src/main/BrowserWinHandler.js:46]: No worries

**Review by tannermpeterson — DISMISSED** [review]

**Review by tannermpeterson — APPROVED** [review]

**Pull request merged**